core/metacling/src/TCling.cxx

```diff
@@ -5891,11 +5867,106 @@ Int_t TCling::AutoParse(const char *cls)
    return nHheadersParsed > 0 ? 1 : 0;
 }
 
+static void* LazyFunctionCreatorAutoloadForModule(const std::string& mangled_name,
+      cling::Interpreter *fInterpreter) {
+   using namespace llvm::object;
+   using namespace llvm::sys::path;
+   using namespace llvm::sys::fs;
+
+   static constexpr std::hash<std::string> hash_fn;
+   // size_t is a hashed mangled_name and libs[(uint8_t)mname_lib_table.second] = library's
+   // name in string.
+   static std::unordered_map<size_t, uint8_t> mname_lib_table;
```

Comment:
There will (almost) always be hash collisions; consider using `unordered_map<size_t,vector<???>>` and then, upon accessing an entry with collision, probably checking where the symbol actually is (o and maybe even warn if it's in multiple libs!).